From 057243803e0a105ef116f2ba2686223fd7968224 Mon Sep 17 00:00:00 2001
From: Ido Ben-Hur <idoybh2@gmail.com>
Date: Mon, 26 Apr 2021 14:49:39 +0300
Subject: [PATCH 082/103] Screenrecord: Save and load set preferences

It's very annoying setting em back for each recording
Save preferences on start, load back when dialog opens

Change-Id: Ic8318cdb72a64da8eeb7d0966d45314d450d2d39
---
 .../ScreenRecordDialogDelegate.java           | 29 +++++++++++++++++++
 1 file changed, 29 insertions(+)

diff --git a/packages/SystemUI/src/com/android/systemui/screenrecord/ScreenRecordDialogDelegate.java b/packages/SystemUI/src/com/android/systemui/screenrecord/ScreenRecordDialogDelegate.java
index ff6a01d6e033..260e67820b60 100644
--- a/packages/SystemUI/src/com/android/systemui/screenrecord/ScreenRecordDialogDelegate.java
+++ b/packages/SystemUI/src/com/android/systemui/screenrecord/ScreenRecordDialogDelegate.java
@@ -44,6 +44,7 @@ import android.widget.TextView;
 import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
 
+import com.android.systemui.Prefs;
 import com.android.systemui.mediaprojection.MediaProjectionCaptureTarget;
 import com.android.systemui.res.R;
 import com.android.systemui.settings.UserContextProvider;
@@ -64,6 +65,13 @@ public class ScreenRecordDialogDelegate implements SystemUIDialog.Delegate {
             MIC_AND_INTERNAL);
     private static final long DELAY_MS = 3000;
     private static final long INTERVAL_MS = 1000;
+    private static final String TAG = "ScreenRecordDialog";
+    private static final String PREFS = "screenrecord_";
+    private static final String PREF_TAPS = "show_taps";
+    private static final String PREF_DOT = "show_dot";
+    private static final String PREF_LOW = "use_low_quality";
+    private static final String PREF_AUDIO = "use_audio";
+    private static final String PREF_AUDIO_SOURCE = "audio_source";
 
     private final SystemUIDialog.Factory mSystemUIDialogFactory;
     private final UserContextProvider mUserContextProvider;
@@ -150,6 +158,8 @@ public class ScreenRecordDialogDelegate implements SystemUIDialog.Delegate {
             }
         });
         mOptions.setLongClickable(false);
+
+        loadPrefs();
     }
 
     /**
@@ -159,6 +169,7 @@ public class ScreenRecordDialogDelegate implements SystemUIDialog.Delegate {
      */
     private void requestScreenCapture(@Nullable MediaProjectionCaptureTarget captureTarget) {
         Context userContext = mUserContextProvider.getUserContext();
+        savePrefs();
         boolean showTaps = mTapsSwitch.isChecked();
         boolean showStopDot = mStopDotSwitch.isChecked();
         boolean lowQuality = mLowQualitySwitch.isChecked();
@@ -196,4 +207,22 @@ public class ScreenRecordDialogDelegate implements SystemUIDialog.Delegate {
             }
         }
     }
+
+    private void savePrefs() {
+        Context userContext = mUserContextProvider.getUserContext();
+        Prefs.putInt(userContext, PREFS + PREF_TAPS, mTapsSwitch.isChecked() ? 1 : 0);
+        Prefs.putInt(userContext, PREFS + PREF_DOT, mStopDotSwitch.isChecked() ? 1 : 0);
+        Prefs.putInt(userContext, PREFS + PREF_LOW, mLowQualitySwitch.isChecked() ? 1 : 0);
+        Prefs.putInt(userContext, PREFS + PREF_AUDIO, mAudioSwitch.isChecked() ? 1 : 0);
+        Prefs.putInt(userContext, PREFS + PREF_AUDIO_SOURCE, mOptions.getSelectedItemPosition());
+    }
+
+    private void loadPrefs() {
+        Context userContext = mUserContextProvider.getUserContext();
+        mTapsSwitch.setChecked(Prefs.getInt(userContext, PREFS + PREF_TAPS, 0) == 1);
+        mStopDotSwitch.setChecked(Prefs.getInt(userContext, PREFS + PREF_DOT, 0) == 1);
+        mLowQualitySwitch.setChecked(Prefs.getInt(userContext, PREFS + PREF_LOW, 0) == 1);
+        mAudioSwitch.setChecked(Prefs.getInt(userContext, PREFS + PREF_AUDIO, 0) == 1);
+        mOptions.setSelection(Prefs.getInt(userContext, PREFS + PREF_AUDIO_SOURCE, 0));
+    }
 }
-- 
2.47.1

